pkg/registry/core/namespace/storage/storage_test.go

```diff
@@ -189,6 +191,143 @@ func TestDeleteNamespaceWithCompleteFinalizers(t *testing.T) {
 	}
 }
 
+func TestDeleteWithGCFinalizers(t *testing.T) {
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.store.DestroyFunc()
+
+	propagationBackground := metav1.DeletePropagationBackground
+	propagationForeground := metav1.DeletePropagationForeground
+	propagationOrphan := metav1.DeletePropagationOrphan
+
+	var tests = []struct {
+		name          string
+		deleteOptions *metav1.DeleteOptions
+
+		existingFinalizers  []string
+		remainingFinalizers map[string]bool
+	}{
+		{
+			name:          "propagation-nil",
+			deleteOptions: nil,
+			existingFinalizers: []string{
+				metav1.FinalizerOrphanDependents,
+				metav1.FinalizerDeleteDependents,
+			},
+			remainingFinalizers: map[string]bool{
+				metav1.FinalizerOrphanDependents: true,
+				metav1.FinalizerDeleteDependents: true,
+			},
+		},
+		{
+			name:                "propagation-nil-without-finalizers",
+			deleteOptions:       nil,
+			existingFinalizers:  []string{},
+			remainingFinalizers: map[string]bool{},
+		},
+		{
+			name: "propagation-background",
+			deleteOptions: &metav1.DeleteOptions{
+				PropagationPolicy: &propagationBackground,
+			},
+			existingFinalizers: []string{
```

Comment:
This should never happen, see https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L205.

Reply:
Done. (removed all the cases where existing finalizers contain both).